src/read/names.rs

```diff
@@ -0,0 +1,2641 @@
+    pub fn die_unit_offset(&self) -> UnitOffset<R::Offset> {
+        self.die_unit_offset
+    }
+
+    /// Resolve this entry to get information about the DIE.
+    ///
+    /// This method validates that the debug_names entry points to a valid DIE
+    /// and returns information about it. For more complex operations that need
+    /// the actual DIE, use `resolve_unit()` to get the unit and then navigate
+    /// to the DIE manually.
+    ///
+    /// # Arguments
+    ///
+    /// * `dwarf` - The main Dwarf structure containing all sections
+    ///
+    /// # Returns
+    ///
+    /// Returns true if the DIE exists and is accessible, false otherwise.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let results = dwarf.find_functions_by_name("main")?;
+    /// for result in results {
+    ///     if result.validate_die(&dwarf)? {
+    ///         let unit = result.resolve_unit(&dwarf)?;
+    ///         // Use unit to access the DIE...
+    ///     }
+    /// }
+    /// ```
+    pub fn validate_die(&self, dwarf: &crate::read::Dwarf<R>) -> Result<bool> {
+        // Get the unit header for this compilation unit
+        let unit_header = dwarf
+            .debug_info
+            .header_from_offset(self.compilation_unit_offset)?;
+
+        // Parse the full unit including abbreviations
+        let unit = dwarf.unit(unit_header)?;
+
+        // Create a cursor and navigate to the specific DIE
+        let mut cursor = unit.entries_at_offset(self.die_unit_offset)?;
+
+        // Check if the DIE exists
+        Ok(cursor.next_dfs()?.is_some())
+    }
+
+    /// Get the compilation unit header for this entry.
+    ///
+    /// This is a lighter-weight alternative to `resolve_die` that only
+    /// returns the unit header without parsing the full unit or DIE.
+    ///
+    /// # Arguments
+    ///
+    /// * `dwarf` - The main Dwarf structure containing all sections
+    pub fn resolve_unit_header(
+        &self,
+        dwarf: &crate::read::Dwarf<R>,
+    ) -> Result<crate::read::UnitHeader<R>> {
+        dwarf
+            .debug_info
+            .header_from_offset(self.compilation_unit_offset)
+    }
+
+    /// Resolve this entry to its full compilation unit.
+    ///
+    /// This method loads and parses the complete compilation unit that
+    /// contains this debug_names entry.
+    ///
+    /// # Arguments
+    ///
+    /// * `dwarf` - The main Dwarf structure containing all sections
+    pub fn resolve_unit(&self, dwarf: &crate::read::Dwarf<R>) -> Result<crate::read::Unit<R>> {
+        let unit_header = self.resolve_unit_header(dwarf)?;
+        dwarf.unit(unit_header)
+    }
+}
+
+/// The header of a `.debug_names` unit.
+#[derive(Debug, Clone)]
+pub struct DebugNamesHeader<R: Reader> {
+    /// Unit length of this name index table.
+    length: R::Offset,
+    /// Version of the name index table format (should be 5 for DWARF 5).
+    version: u16,
+    /// Padding for alignment.
+    padding: u16,
```

Comment:
Why is the padding here? This is not `repr(C)`, so we aren't trying to zero copy parse this data.

The spec says:

> padding (uhalf)
> Reserved to DWARF (must be zero).